hashview/models.py

```diff
@@ -21,24 +24,78 @@ class Users(db.Model, UserMixin):
     admin = db.Column(db.Boolean, nullable=False, default=False)
     pushover_app_id = db.Column(db.String(50), nullable=True)
     pushover_user_key = db.Column(db.String(50), nullable=True)
+    last_login_utc = db.Column(db.DateTime, nullable=True, default=datetime.utcnow)
     wordlists = db.relationship('Wordlists', backref='tbd', lazy=True)
     rules = db.relationship('Rules', backref='owner', lazy=True)
     jobs = db.relationship('Jobs', backref='owner', lazy=True)
     tasks = db.relationship('Tasks', backref='owner', lazy=True)
     taskgroups = db.relationship('TaskGroups', backref='owner', lazy=True)
 
-    def get_reset_token(self, expires_sec=1800):
-        s = Serializer(Config.SECRET_KEY, expires_sec)
-        return s.dumps({'user_id': self.id}).decode('utf-8')
+    def _get_reset_token_salt(self) -> str:
+        """
+        Create salt data for password reset token signing. The return value will be hashed
+        together with the signing key. This ensures that changes to any of the fields included
+        in the salt invalidates any tokens produced with the old values.
+        """
+        return json.dumps([
+            self.first_name,
+            self.last_name,
+            self.password.hash.decode() if self.password is not None else '',
```

Comment:
So... i think this _should_ be 
`self.password if self.password is not None else '',`

Not entirely sure though

Reply:
Yup. You're absolutely right. No need to use anything other than the raw string that the password is stored as.